tests/testthat/test-scale-discrete.R

```diff
@@ -16,3 +16,19 @@ test_that("discrete ranges also encompas continuous values", {
   expect_equal(x_range(base + geom_point(aes(x1)) + geom_point(aes(x2))), c(0, 4))
 })
 
+test_that("discrete scale shrinks to range when setting limits", {
+  p <- ggplot(subset(diamonds, carat > 1), aes(cut, clarity)) +
+    geom_jitter() + scale_x_discrete(limits=c("Fair","Ideal"))
+
+  expect_equal(layer_scales(p)$x$dimension(c(0, 1)), c(0, 3))
+})
+
+test_that("limits are returned correctly when using continuous input", {
+  xyDF <- data.frame(x = 1:10, y = 1:10)
+  p <- ggplot(data = xyDF, mapping = aes(x = x, y = y)) +
+    geom_line() +
+    geom_point() +
+    scale_x_discrete()
+
+  expect_equal(layer_scales(p)$x$get_limits(), seq_len(10))
```

Comment:
Yeah(-ish), the git history was not completely transparent but it was partly in response to #1542. Problem was that it broke everything else :-)

As far as I recall we decided to rollback this change as it was not obvious how a discrete scale should behave in the face of an enormous number of levels...

I would suggest we fix this be adding a `max-breaks` to ScaleDiscrete instead, as this would make it more transparent what goes on inside the scale...